### ISSUE TITLE
Executing of sbt "runMain my.package.Class" on windows will fail

### ISSUE THREAD

**Issue by HarryEuro** [sbt/sbt#5222]
## steps
sbt version: 1.3.3

## problem
Executing of `sbt.bat` will fail on commands like this: `sbt "runMain my.package.Class"`.
Error: 
```
my.package.Class was unexpected at this time.
```

## expectation
Executing of `sbt "runMain my.package.Class"` should be working.

## notes
The problem is the batch script `sbt.bat` starting on line 383:
```
if /I "%g:~0,2%" == "-D" (
 rem special handling for -D since '=' gets parsed away
 if x%g:^==% == x%g% (
   if not "%~1" == "" (
     set SBT_ARGS=!SBT_ARGS! %0=%1
     shift
     goto args_loop
   ) else (
     echo %g is missing a value
     goto error
   )
 )
)
```

After removing this piece of code the execution of this command `sbt "runMain my.package.Class"` works.

**Comment by eed3si9n** [comment]
@HarryEuro Thanks for the report!

**Comment by thebridsk** [comment]
The problem is the line:

     if x%g:^==% == x%g% (

quotes are needed

     if "x%g:^==%" == "x%g%" (

**Comment by eed3si9n** [comment]
Looks like I fixed it in https://github.com/sbt/sbt-launcher-package/pull/300

**Comment by XavierGeerinck** [comment]
Note: runMain is again not working on version 1.3.5, had to rollback to 1.3.3 and apply the replacement fix by @thebridsk

**Comment by thebridsk** [comment]
@thebillkidy There is a fix that may solve your problem on 1.3.5.  See sbt/sbt-launcher-package#309
This fix worked for me.
